cachi2/core/package_managers/npm.py

```diff
@@ -476,6 +502,104 @@ def _get_npm_dependencies(
+    :param output_dir: Output directory with fetched dependencies
+    """
+    package_lock = PackageLock.from_file(RootedPath(package_lock_file.abspath))
+
+    if package_lock.lockfile_version > 1:
+        for package in package_lock._get_packages():
+            # Remove integrity for git sources, their integrity checksum will change when
+            # constructing tar archive from cloned repository
+            url = _normalize_resolved_url(package.resolved_url)
+            if _classify_resolved_url(url) == "git":
+                if package.integrity:
+                    package.integrity = ""
+
+            # Update all type of dependencies except file:
+            if _classify_resolved_url(url) != "file":
+                templated_abspath = str(download_paths[url].path).replace(
+                    str(output_dir.path), "${output_dir}"
+                )
+                package.resolved_url = f"file://{templated_abspath}"
+
+            if package.dependencies:
+                for dependency, dependency_version in package.dependencies.items():
+                    if get_type_of_dependency(dependency_version) in ["https", "git"]:
+                        package.dependencies.update({dependency: ""})
+
+        # TODO: Tried to use get_main_package method but it failed horribly with error:
+        #   target is outside `package-lock.json` Error: PathOutsideRoot: Joining path '..' to `package-lock.json` target is outside
+        main_package = package_lock._lockfile_data["packages"][""]
+        if main_package["dependencies"]:
+            for dependency, dependency_version in main_package["dependencies"].items():
+                if get_type_of_dependency(dependency_version) in ["https", "git"]:
+                    main_package["dependencies"].update({dependency: ""})
+
+    return ProjectFile(abspath=package_lock_file.abspath, template=str(package_lock._lockfile_data))
+
+
+def update_package_json_files(workspaces: list[str], source_dir: RootedPath) -> list[ProjectFile]:
+    """Set dependencies to empty string in package.json files.
+
+    :param workspaces: list of workspaces paths
+    :param source_dir: Source directory with repository content
+    """
+    package_json_paths = []
+    for workspace in workspaces:
+        package_json_paths.append(source_dir.join_within_root(workspace, "package.json"))
+    package_json_paths.append(source_dir.join_within_root("package.json"))
+
+    package_json_projectfiles = []
+    for package_json_path in package_json_paths:
+        with open(package_json_path, "r") as f:
+            package_json_content = json.load(f)
+        for dependency, url in package_json_content["dependencies"].items():
+            if get_type_of_dependency(url) in ["https", "git"]:
+                package_json_content["dependencies"].update({dependency: ""})
+
+        package_json_projectfiles.append(
+            ProjectFile(abspath=package_json_path.path, template=str(package_json_content))
```

Comment:
we should `json.dumps` the content, the str() of a dict is not valid JSON

Reply:
`json.dumps` used